lib/rpmte.cc

```diff
@@ -151,17 +151,37 @@ static int addTE(rpmte p, Header h, fnpyKey key, rpmRelocation * relocs)
     p->release = headerGetAsString(h, RPMTAG_RELEASE);
 
     /* name, version and release are required in all packages */
-    if (p->name == NULL || p->version == NULL || p->release == NULL)
-	goto exit;
+    for (auto [value, name] : std::vector<std::pair<char *, std::string>>
+	     {{p->name, "name"},
+	      {p->version, "version"},
+	      {p->release, "release"}}) {
+	if (value == NULL) {
+	    rpmlog(RPMLOG_ERR, _("Package is missing %s\n"), name.c_str());
+	    goto exit;
```

Comment:
Nah, just leave this like it was, and with a central error message at the end if you like. These are not "packager did something wrong" situations where the user could fix something, it's just a malformed package.

Reply:
Hey, I was sooo proud of getting this loop to actually work...